implementations/rust/ockam/ockam_entity/src/channel/listener.rs

```diff
@@ -1,28 +1,18 @@
-use crate::{ProfileImpl, ProfileVault, Responder};
+use crate::{ProfileTrait, ProfileVault, Responder};
 use ockam_channel::{CreateResponderChannelMessage, SecureChannel};
 use ockam_core::{Address, Result, Routed, Worker};
 use ockam_key_exchange_xx::XXNewKeyExchanger;
 use ockam_node::Context;
 use rand::random;
 
-pub(crate) struct ProfileChannelListener<V: ProfileVault> {
-    profile: ProfileImpl<V>, // TODO: Avoid copying profile
-    vault: V,
-    listener_address: Option<Address>,
-}
-
-impl<V: ProfileVault> ProfileChannelListener<V> {
-    pub fn new(profile: ProfileImpl<V>, vault: V) -> Self {
```

Comment:
This was making things complicated when using the generics, and constructing it inline was easier. Can't remember which type was giving trouble.